services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtResponseHandlerAdapter.java

```diff
@@ -46,9 +48,14 @@ public final class S3CrtResponseHandlerAdapter implements S3MetaRequestResponseH
     private final SdkHttpResponse.Builder respBuilder = SdkHttpResponse.builder();
     private volatile S3MetaRequest metaRequest;
 
-    public S3CrtResponseHandlerAdapter(CompletableFuture<Void> executeFuture, SdkAsyncHttpResponseHandler responseHandler) {
+    private final PublisherListener<S3MetaRequestProgress> progressListener;
```

Comment:
Do we need to invoke `this.progressListener.subscriberOnError();` in `onErrorResponseComplete`?

Reply:
`failResponseHandlerAndFuture` doesn't seem to be invoked from `onErrorResponseComplete` if it successfully marshalls the error response though. Can we add a test to verify it? [This test](https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerListenerTest.java#L65) may not suffice now that we are relying on the real CRT client to report progress and it is using a mock s3 client.

```
        responsePublisher.send(ByteBuffer.wrap(errorPayload))
                         .thenRun(responsePublisher::complete)
                         .handle((ignore, throwable) -> {
                             if (throwable != null) {
                                 failResponseHandlerAndFuture(throwable);
                                 return null;
                             }
                             completeFutureAndCloseRequest();
                             return null;
                         });
```

Reply:
Added new mock tests and removed  explicit onError